bloomfilters/matt-keibler/test/MockHashCalculator.cpp

```diff
@@ -0,0 +1,27 @@
+#include "MockHashCalculator.h"
+
+MockHashCalculator::MockHashCalculator() : HashCalculator(), failed(false) {
+
+}
+
+Hash MockHashCalculator::calculate(string word){
+  if(!failed && !to_hash.empty() && ! to_return.empty()){
+    if(word == to_hash.front()){
+      to_hash.pop();
+      Hash return_value = to_return.front();
+      to_return.pop();
+      return return_value;
+    }
+    failed = true;
+  }
+  failed = true;
+}
+
+void MockHashCalculator::add_verify_and_return(string word, Hash hash){
```

Comment:
Google has one. I have never used it, so I went with pure C++ for the quick and simple case. I would probably go back and update the test to use that library in the future as part of a refactor pass.